samples/cellular/nrf_cloud_mqtt_cell_location/src/main.c

```diff
@@ -0,0 +1,637 @@
+		break;
+	case NET_EVENT_L4_DISCONNECTED:
+		LOG_INF("Network connectivity lost");
+		break;
+	default:
+		LOG_DBG("Unknown event: 0x%08llX", event);
+		return;
+	}
+}
+
+/* Callback to track connectivity layer events */
+static struct net_mgmt_event_callback conn_cb;
+static void connectivity_event_handler(struct net_mgmt_event_callback *cb,
+				       uint64_t event,
+				       struct net_if *iface)
+{
+	if (event == NET_EVENT_CONN_IF_FATAL_ERROR) {
+		LOG_ERR("NET_EVENT_CONN_IF_FATAL_ERROR");
+		return;
+	}
+}
+
+/* Start tracking network availability */
+static int prepare_network_tracking(void)
+{
+	net_mgmt_init_event_callback(&l4_callback, l4_event_handler, L4_EVENT_MASK);
+	net_mgmt_add_event_callback(&l4_callback);
+
+	net_mgmt_init_event_callback(&conn_cb, connectivity_event_handler, CONN_LAYER_EVENT_MASK);
+	net_mgmt_add_event_callback(&conn_cb);
+
+	return 0;
+}
+
+SYS_INIT(prepare_network_tracking, APPLICATION, 0);
+
+static void cloud_service_location_ready_cb(const struct nrf_cloud_location_result *const result)
+{
+	if (!result) {
+		return;
+	}
+
+	if (result->err == NRF_CLOUD_ERROR_NONE) {
+		const char *type_str = (result->type == LOCATION_TYPE_SINGLE_CELL) ? "single-cell" :
+				       (result->type == LOCATION_TYPE_MULTI_CELL) ? "multi-cell" :
+										     "unknown";
+		LOG_INF("Cellular location request fulfilled with %s", type_str);
+		LOG_INF("Lat: %f, Lon: %f, Uncertainty: %u m", result->lat, result->lon,
+			result->unc);
+	} else {
+		LOG_ERR("Unable to determine location from cloud service, error: %d", result->err);
+	}
+}
+
+int main(void)
+{
+	int err;
+
+	LOG_INF("nRF Cloud MQTT Cellular Location Sample, version: %s", APP_VERSION_STRING);
+
+	err = init();
+	if (err) {
+		LOG_ERR("Initialization failed");
+		return 0;
+	}
+
+	/* Connect to nRF Cloud */
+	err = connect_cloud();
+	if (err) {
+		LOG_ERR("Cloud connection failed");
+		return 0;
+	}
+
+	/* Send device status with HW/FW version and network details */
+	if (IS_ENABLED(CONFIG_MQTT_CELL_SEND_DEVICE_STATUS)) {
+		send_device_status();
+	}
+
+	/* Get initial cell info */
+	get_cell_info();
+
+	/* Main loop: process cell location requests */
+	while (1) {
+		/* Wait for cell info (triggered by button press or cell ID change) */
+		(void)k_sem_take(&cell_info_ready_sem, K_FOREVER);
+
+		(void)k_mutex_lock(&cell_info_mutex, K_FOREVER);
+
+		/* Log current cell information */
+		if (cell_info.current_cell.id != LTE_LC_CELL_EUTRAN_ID_INVALID) {
+			LOG_INF("Current cell info: Cell ID: %u, TAC: %u, MCC: %d, MNC: %d",
+				cell_info.current_cell.id, cell_info.current_cell.tac,
+				cell_info.current_cell.mcc, cell_info.current_cell.mnc);
+		} else {
+			LOG_WRN("No current serving cell available");
+		}
+
+		/* Log request type */
+		if (cell_info.ncells_count || cell_info.gci_cells_count) {
+			LOG_INF("Performing multi-cell request with "
+				"%u neighbor cells and %u GCI cells",
+				cell_info.ncells_count, cell_info.gci_cells_count);
```

Comment:
At some point, cloud team counted only GCI cells as multicell location. Current cell and neighbor cells are for single cell positioning. This was extensively investigated by cloud team 2 or 3 years ago.
I do not know if this plan has changed.